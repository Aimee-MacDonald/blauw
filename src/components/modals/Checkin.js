import React from 'react'
import {connect} from 'react-redux'

import {setModal} from '../../state_management/actions/modal'

export const Checkin = props => (
  <div>
    <h1>Check this booking in?</h1>
    <button>Yes</button>
    <button onClick={() => props.dispatch(setModal())}>No</button>
  </div>
)

export default connect()(Checkin)