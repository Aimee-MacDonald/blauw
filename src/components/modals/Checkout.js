import React, {useContext} from 'react'
import {connect} from 'react-redux'

import {ServerConnectionContext} from '../../util/ServerConnection'

import {checkOut} from '../../state_management/actions/bookings'
import {setModal} from '../../state_management/actions/modal'
import {getSelectedBooking} from '../../state_management/selectors/bookings'

export const Checkout = props => {
  const server = useContext(ServerConnectionContext)

  const checkBookingOut = bookingId => {
    props.dispatch(checkOut(bookingId))
    server.dispatch({reducer: 'bookings', action: checkOut(bookingId)})
    props.dispatch(setModal())
  }

  return(
    <div>
      <h1>Check this booking out?</h1>
      <button onClick={() => checkBookingOut(props.booking._id)}>Yes</button>
      <button onClick={() => props.dispatch(setModal())}>No</button>
    </div>
  )
}

const mapStateToProps = ({bookings}) => ({booking: getSelectedBooking(bookings)})

export default connect(mapStateToProps)(Checkout)