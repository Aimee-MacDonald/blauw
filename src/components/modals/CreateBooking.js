import React, {useContext} from 'react'
import styled from 'styled-components'
import {connect} from 'react-redux'

import {ServerConnectionContext} from '../../util/ServerConnection'

import {setModal} from '../../state_management/actions/modal'
import {createBooking} from '../../state_management/actions/bookings'

const CreateBooking = props => {
  const connection = useContext(ServerConnectionContext)

  const saveBooking = e => {
    e.preventDefault()

    const bookingDetails = {
      name: e.target.name.value,
      date: parseInt(e.target.date.value),
      room: parseInt(e.target.room.value),
      nights: parseInt(e.target.nights.value)
    }

    const createBookingAction = createBooking(bookingDetails)

    props.dispatch(createBookingAction)
    connection.dispatch({reducer: 'bookings', action: createBookingAction})
    props.dispatch(setModal())
  }

  return(
    <StyledCreateBooking onSubmit={saveBooking}>
      <h1>Create Booking</h1>

      <label htmlFor='name'>Name:</label>
      <input id='name' />

      <label htmlFor='date'>Date:</label>
      <input id='date' />

      <label htmlFor='room'>Room:</label>
      <input id='room' />

      <label htmlFor='nights'>Nights:</label>
      <input id='nights' />
      
      <button type='reset' onClick={() => props.dispatch(setModal())}>Cancel</button>
      <button type='submit'>Save</button>
    </StyledCreateBooking>
  )
}

const StyledCreateBooking = styled.form`
  background-color: lime;
`

export default connect()(CreateBooking)