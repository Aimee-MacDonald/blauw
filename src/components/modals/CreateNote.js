import React from 'react'
import styled from 'styled-components'
import {connect} from 'react-redux'

import {setModal} from '../../state_management/actions/modal'
import {createNote} from '../../state_management/actions/notes'

export const CreateNote = props => {
  const addNote = e => {
    e.preventDefault()

    const note = {
      title: e.target.title.value,
      content: e.target.content.value
    }

    props.dispatch(createNote(note))
    props.dispatch(setModal())
  }

  return(
    <StyledCreateNote onSubmit={addNote}>
      <h1>Create Note</h1>

      <label htmlFor='title'>Title</label>
      <input id='title'/>

      <textarea id='content'/>

      <button type='button' onClick={() => props.dispatch(setModal())}>Cancel</button>
      <button type='submit'>Save</button>
    </StyledCreateNote>
  )
}

const StyledCreateNote = styled.form`
  display: grid;
  grid-template-columns: 1fr 1fr;

  h1, textarea{
    grid-column-start: 1;
    grid-column-end: 3;
  }
`

export default connect()(CreateNote)