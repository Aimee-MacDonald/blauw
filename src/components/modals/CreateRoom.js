import React, {useContext} from 'react'
import {connect} from 'react-redux'
import styled from 'styled-components'

import {ServerConnectionContext} from '../../util/ServerConnection'
import {createRoom} from '../../state_management/actions/rooms'

import {setModal} from '../../state_management/actions/modal'

export const CreateRoom = props => {
  const server = useContext(ServerConnectionContext)

  const saveRoom = e => {
    e.preventDefault()

    const createRoomAction = createRoom({
      name: e.target.name.value,
      group: e.target.group.value,
      shared: e.target.shared.checked,
      maxPax: e.target.maxPax.value,
      basePrice: e.target.basePrice.value,
      personPrice: e.target.personPrice.value,
      showBeds: e.target.showBeds.checked
    })

    props.dispatch(createRoomAction)
    server.dispatch({reducer: 'rooms', action: createRoomAction})
    props.dispatch(setModal())
  }

  return(
    <StyledCreateRoom onSubmit={saveRoom}>
      <h1>Add Room</h1>

      <label htmlFor='name'>Name</label>
      <input id='name'/>

      <label htmlFor='group'>Group</label>
      <input id='group'/>

      <label htmlFor='shared'>Shared</label>
      <input id='shared' type='checkbox'/>

      <label htmlFor='maxPax'>Max Pax</label>
      <input id='maxPax'/>

      <label htmlFor='basePrice'>Base Price</label>
      <input id='basePrice'/>

      <label htmlFor='personPrice'>Person Price</label>
      <input id='personPrice'/>

      <label htmlFor='showBeds'>Show Beds</label>
      <input id='showBeds' type='checkbox'/>

      <button type='reset' onClick={() => props.dispatch(setModal())}>Cancel</button>
      <button type='submit'>Save</button>
    </StyledCreateRoom>
  )
}

const StyledCreateRoom = styled.form`
  display: grid;
  grid-template-columns: 1fr 1fr;
  padding: 1.6rem;

  h1{
    grid-column-start: 1;
    grid-column-end: 3;
    margin-bottom: 3.2rem;
    text-align: center;
  }

  label{
    text-align: right;
    margin-top: 0.4rem;
    padding-right: 0.8rem;
  }

  input{
    background-color: transparent;
    border: none;
    border-bottom 1px solid black;
    width: 75%;
    margin-top: 0.4rem;
  }

  button{
    padding: 0.4rem;
    margin: 3.2rem 0.1rem 0 0.2rem;
  }
`

export default connect()(CreateRoom)