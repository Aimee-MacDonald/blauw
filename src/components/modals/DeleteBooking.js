import React, {useContext} from 'react'
import {connect} from 'react-redux'

import {ServerConnectionContext} from '../../util/ServerConnection'

import {deleteBooking} from '../../state_management/actions/bookings'
import {setModal} from '../../state_management/actions/modal'

const DeleteBooking = props => {
  const connection = useContext(ServerConnectionContext)

  return(
    <div>
      <h1>Delete this booking?</h1>
      <p>{props._id}</p>
      <button onClick={() => {
        props.dispatch(deleteBooking(props._id))
        connection.dispatch({reducer: 'bookings', action: deleteBooking(props._id)})
        props.dispatch(setModal())
      }}>Yes</button>
      <button onClick={() => props.dispatch(setModal())}>No</button>
    </div>
  )
}

export default connect()(DeleteBooking)