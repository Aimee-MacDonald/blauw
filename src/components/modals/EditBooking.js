import React, {useState, useContext} from 'react'
import styled from 'styled-components'
import {connect} from 'react-redux'

import {ServerConnectionContext} from '../../util/ServerConnection'

import {updateBooking} from '../../state_management/actions/bookings'
import {setModal} from '../../state_management/actions/modal'

const EditBooking = props => {
  const [bookingDetails, editBooking] = useState(props)
  const connection = useContext(ServerConnectionContext)

  const saveBooking = e => {
    e.preventDefault()
    
    props.dispatch(updateBooking(bookingDetails))
    connection.dispatch({reducer: 'bookings', action: updateBooking(bookingDetails)})
    props.dispatch(setModal())
  }

  return(
    <StyledEditBooking onSubmit={saveBooking}>

      <h1>Edit Booking</h1>

      <label htmlFor='name'>Name:</label>
      <input id='name' value={bookingDetails.name} onChange={e => editBooking({...bookingDetails, name: e.target.value})}/>

      <label htmlFor='date'>Date:</label>
      <input id='date' value={bookingDetails.date} onChange={e => editBooking({...bookingDetails, date: parseInt(e.target.value) || 0})}/>

      <label htmlFor='room'>Room:</label>
      <input id='room' value={bookingDetails.room} onChange={e => editBooking({...bookingDetails, room: parseInt(e.target.value) || 0})}/>

      <label htmlFor='nights'>Nights:</label>
      <input id='nights' value={bookingDetails.nights} onChange={e => editBooking({...bookingDetails, nights: parseInt(e.target.value) || 0})}/>

      <button type='reset' onClick={() => props.dispatch(setModal())}>Cancel</button>
      <button type='submit'>Save</button>
    </StyledEditBooking>
  )
}

const StyledEditBooking = styled.form`
  display: grid;
  grid-template-columns: 1fr 1fr;
  padding: 1.6rem;

  h1{
    grid-column-start: 1;
    grid-column-end: 3;
    margin-bottom: 3.2rem;
    text-align: center;
  }

  label{
    text-align: right;
    margin-top: 0.4rem;
    padding-right: 0.8rem;
  }

  input{
    background-color: transparent;
    border: none;
    border-bottom 1px solid black;
    width: 75%;
    margin-top: 0.4rem;
  }

  button{
    padding: 0.4rem;
    margin: 3.2rem 0.1rem 0 0.2rem;
  }
`

const mapStateToProps = ({bookings}) => bookings.bookings.filter(booking => booking.id === bookings.selectedBooking)[0]
export default connect(mapStateToProps)(EditBooking)