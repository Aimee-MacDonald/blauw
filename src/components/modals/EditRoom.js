import React from 'react'
import {connect} from 'react-redux'

import {setModal} from '../../state_management/actions/modal'

export const EditRoom = props => (
  <div>
    <h1>Edit Room</h1>

    <button onClick={() => props.dispatch(setModal())}>Cancel</button>
    <button>Save</button>
  </div>
)

export default connect()(EditRoom)