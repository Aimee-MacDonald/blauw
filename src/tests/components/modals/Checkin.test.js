import React from 'react'
import {shallow} from 'enzyme'

import {Checkin} from '../../../components/modals/Checkin'

test('Render Checkin Modal', () => {
  const wrapper = shallow(<Checkin/>)
  expect(wrapper.getElement()).toMatchSnapshot()
})