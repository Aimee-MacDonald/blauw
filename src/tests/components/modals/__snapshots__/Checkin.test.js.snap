// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Checkin Modal 1`] = `
<div>
  <h1>
    Check this booking in?
  </h1>
  <button>
    Yes
  </button>
  <button
    onClick={[Function]}
  >
    No
  </button>
</div>
`;
